Actions/GetArtifactsForDeployment/GetArtifactsForDeployment.psm1

```diff
@@ -0,0 +1,340 @@
+                break
+            }
+
+            if ($latestPrRun.conclusion -ne 'success') {
+                Write-Host "::error::Latest PR build run is not successful. ($($latestPrRun.html_url))"
+                break
+            }
+            # We only care about the latest PR build. If it is not completed or not successful, we return 0
+            $lastSuccessfulPRRun = $latestPrRun.id
+            $lastKnownGoodBuildId = FindPRRunAnnotationForIncrementalBuilds -repository $repository -checkSuiteId $latestPrRun.check_suite_id -token $token
+            break
+        }
+
+        if($lastSuccessfulPRRun -ne 0) {
+            break
+        }
+
+        $page += 1
+    }
+
+    if($lastSuccessfulPRRun -ne 0) {
+        Write-Host "Lastest PR build ($lastSuccessfulPRRun) was successful for branch $branch in repository $repository"
+    } else {
+        Write-Host "Lastest PR build ($lastSuccessfulPRRun) was not successful for branch $branch in repository $repository"
+    }
+
+    return $lastSuccessfulPRRun, $lastKnownGoodBuildId
+}
+
+<#
+    .SYNOPSIS
+    Downloads an artifact from a workflow run and unpacks it.
+    .PARAMETER token
+    Auth token
+    .PARAMETER folder
+    Folder to download the artifact to
+    .PARAMETER artifact
+    The artifact to download
+#>
+function DownloadAndUnpackArtifact {
+    Param(
+        [string]$token,
+        [string]$folder,
+        $artifact
+    )
+
+    if ([string]::IsNullOrEmpty($token)) {
+        $token = invoke-gh -silent -returnValue auth token
+    }
+    $headers = GetHeaders -token $token
+
+    $filename = "$folder.zip"
+    InvokeWebRequest -Headers $headers -Uri $artifact.archive_download_url -OutFile $filename
+    if (Test-Path $folder) {
+        Remove-Item $folder -Recurse -Force
+    }
+    Expand-Archive -Path $filename -DestinationPath $folder
+    Remove-Item $filename -Force
+}
+
+<#
+    .SYNOPSIS
+    Downloads artifacts from a PR build and related last known good build.
+    Iterates through the last known good build, and only copies any apps not found in the PR build.
+
+    Any apps copied from the last known good build, are copied to the related PR artifact folder, based on project and artifact type.
+    In case a PR artifact folder does not exist for a specific app, it is created.
+    This is necessary, since the deploy action expects artifacts to follow the PR naming convention when deploying from a PR.
+    .PARAMETER token
+    Auth token
+    .PARAMETER path
+    Path to the build artifacts folder
+    .PARAMETER prArtifacts
+    List of artifacts published by the PR build
+    .PARAMETER lastKnownGoodBuildArtifacts
+    List of artifacts published by the last known good build, linked from the PR build
+    .PARAMETER prRunId
+    Run id of the PR build
+    .PARAMETER lastKnownGoodBuildRunId
+    Run id of the last known good build
+#>
+function DownloadPRArtifacts {
+    Param(
+        [string]$token,
```

Comment:
```suggestion
        [Parameter(Mandatory = $true)]
        [string]$token,
```

Reply:
Please also add a space between the type and the variable name - we do that all over.